### PR TITLE
Improve DirectoryDAOTest::relocateDirectory()

### PR DESCRIPTION
**Pull request by uklotzde**
As requested: https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/cmake.20on.20macOS/near/216786005

Not sure if this fixes the actual issues, but it already contains some changes from a future PR. If it doesn't fix the macOS issues it is still an improvement with additional checks during the test run.

### CODE REVIEW

**Comment by Be-ing** [comment]
I tried cherry-picking this to #3305 but the test still fails: https://github.com/mixxxdj/mixxx/pull/3305/checks?check_run_id=1402974906#step:5:2483

**Comment by uklotzde** [comment]
> I tried cherry-picking this to #3305 but the test still fails: https://github.com/mixxxdj/mixxx/pull/3305/checks?check_run_id=1402974906#step:5:2483

Unfortunately I am not able to see the results of the failing tests. The link points to a run with tests disabled.

**Comment by Be-ing** [comment]
Any other ideas? I'm not familiar with the DirectoryDAO code at all.

**Comment by uklotzde** [comment]
No. Without the ability to reproduce the bug in a local development chasing this bug would be time consuming.

You don't need to disable all tests, only the failing test: `--exclude-regex DirectoryDAOTest.relocateDirectory`

File: src/test/directorydaotest.cpp [hunk 1]
```diff
@@ -130,16 +130,25 @@ TEST_F(DirectoryDAOTest, getDirTest) {
...
-    // use a temp dir so that we always use a real existing system path
-    QString testdir(QDir::tempPath() + "/TestDir");
-    QString test2(QDir::tempPath() + "/TestDir2");
-    QString testnew(QDir::tempPath() + "/TestDirNew");
+    //create temp dirs
+    QString testdir(tempDir1.path() + "/TestDir");
+    ASSERT_TRUE(QDir(tempDir1.path()).mkpath(testdir));
+    QString test2(tempDir2.path() + "/TestDir2");
+    ASSERT_TRUE(QDir(tempDir2.path()).mkpath(test2));
+    QString testnew(tempDir2.path() + "/TestDirNew");
```
Comment (Holzhaus) [inline src/test/directorydaotest.cpp:145]:
```suggestion
    QString testdir(tempDir1.filePath("TestDir"));
    ASSERT_TRUE(QDir(tempDir1.path()).mkpath(testdir));
    QString test2(tempDir2.filePath("TestDir2"));
    ASSERT_TRUE(QDir(tempDir2.path()).mkpath(test2));
    QString testnew(tempDir2.filePath("TestDirNew"));
```

**Review by Holzhaus — APPROVED** [review]
LGTM, waiting for CI.

**Pull request merged**